credit.py

```diff
@@ -7,16 +7,27 @@
                     format='%(asctime)s %(message)s', 
                     filemode='w') 
 logger = logging.getLogger() 
-logger.setLevel(logging.ERROR) 
+logger.setLevel(logging.WARNING) 
 
 def load_credits(path_credit):
 
     try:
         with open(path_credit) as f:
             credit = json.load(f)
-            if type(credit) is not dict:
+            if not isinstance(credit, dict):
                 logger.error('Data in credentials.txt is not dictionary')
-                return None
+                raise TypeError
+            k = 0
+            v = 0
+            for key in credit:
+                if not isinstance(key, str):
+                    k += 1
+            for value in credit:
+                if not isinstance(value, str):
+                    v += 1
+            if k > 0 or v > 0:
+                logger.error(TypeError('%d keys and %d values in credit dictionary are not string' % (k, v)))
+                raise TypeError
```

Comment:
You return `None` in for other exceptional situations, please do it here as well.